### DISCUSSION TITLE
On the same line align one block far left and another block far right with centered spacing

### DISCUSSION THREAD

**Discussion by kalensk** [charmbracelet/lipgloss#227] [Q&A]
Is there a way to render text as follows with block1 hugging the left margin and block2 hugging the right margin? I've tried a bunch of things, but nothing seems to get it.

```
[ block1Text               block2Text ]
```

Technically I have several blocks of text and trying to get something like the following:
```
-----------------------------------------------------
| leftHeader                            rightHeader |
| 1 cell padding                                    |
| MainBlockOfText                                   |
|                                                   |
------------------------------------------------------
```

So far Im trying something like the following:
```go
boarder := lipgloss.NewStyle().
		Width(120).
		Border(lipgloss.NormalBorder(), true, true, true, true)

	headerStyle := lipgloss.NewStyle().Width(120).PaddingBottom(1)
	leftHeader := lipgloss.NewStyle().AlignHorizontal(lipgloss.Left)
	rightHeader := lipgloss.NewStyle().AlignHorizontal(lipgloss.Right)
	header := headerStyle.Render(leftHeader.Render("left header") + rightHeader.Render("right header line1\n right header line 2"))

	return boarder.Render(
		lipgloss.JoinVertical(lipgloss.Left,
			header,
			model.View(),
		),
	)
```

**Comment by kalensk** [comment]
After much struggling I got it mostly working with:

```go
boarder := lipgloss.NewStyle().
		Width(120).
		Border(lipgloss.NormalBorder(), true, true, true, true)

	leftHeader1 := lipgloss.NewStyle().Width(60).Align(lipgloss.Left).Render("left header line1")
	rightHeader1 := lipgloss.NewStyle().Width(60).Align(lipgloss.Right).Render("right header text line1")
	rightHeader2 := lipgloss.NewStyle().Width(120).Align(lipgloss.Right).PaddingBottom(1).Render("right header text line2")
	header := leftHeader1 + rightHeader1 + "\n" + rightHeader2

	return boarder.Render(
		lipgloss.JoinVertical(lipgloss.Left,
			header,
			model.View(),
		),
	)
```

However, if I want to adding padding to the boarder so things are not pushing against it, everything falls apart...Its all so fragile!

**Comment by beltagyy** [comment]
To align one block to the left and another to the right inside a fixed-width container, use lipgloss.Place.

Pattern:
- Create a parent style with a fixed width (e.g., 120).
- Render the left and right blocks separately.
- Use lipgloss.JoinHorizontal with lipgloss.Top alignment.
- Or compute the gap manually: measure the rendered width of left and right, then insert enough spaces between them so the total equals the parent width minus borders.

A concise way if the widths are known:
  parent := lipgloss.NewStyle().Width(120)
  left := lipgloss.NewStyle().AlignHorizontal(lipgloss.Left).Render("left header")
  right := lipgloss.NewStyle().AlignHorizontal(lipgloss.Right).Render("right header line 1\n right header line 2")
  row := lipgloss.JoinHorizontal(lipgloss.Top, left, right)

If the combined width is less than the parent, lipgloss.JoinHorizontal will leave a natural gap because each block carries its own width. Make sure each block style has an explicit Width set so lipgloss knows how much space to allocate for alignment.
